packages/babel-plugin/src/tests/generator/variants.ts

```diff
@@ -127,12 +129,12 @@ function unionUndefined(type: string | undefined, optional: OptionalVariant) {
 // TODO: Consider generating variants for optional properties where both question mark and undefined before/after type are used
 export function generatePropertyVariants(options: PropertyTestOptions): PropertyVariant[] {
   const variants: PropertyVariant[] = [];
-  const typeVariants = generateTypeNameVariants(options.type);
+  const typeVariants = generateTypeNameVariants(options.type !== "object" ? options.type : options.objectType);
   if (DEFAULT_INFERABLE_TYPES.has(options.type) && typeof options.default !== "undefined") {
     typeVariants.push(undefined);
   }
   for (const type of typeVariants) {
-    for (const typeArgument of generateTypeNameVariants(options.objectType)) {
+    for (const typeArgument of generateTypeNameVariants(options.type !== "object" ? options.objectType : undefined)) {
```

Comment:
Same as previous comment.
```suggestion
    for (const typeArgument of generateTypeNameVariants(options.type === "object" ? undefined : options.objectType)) {
```